policyengine_us/variables/gov/hhs/medicaid/eligibility/is_medicaid_eligible.py

```diff
@@ -6,8 +6,14 @@ class is_medicaid_eligible(Variable):
     entity = Person
     label = "Eligible for Medicaid"
     definition_period = YEAR
-    reference = "https://www.law.cornell.edu/uscode/text/42/1396a#a_10"
+    reference = (
+        "https://www.law.cornell.edu/uscode/text/42/1396a#a_10"
+        "https://www.kff.org/racial-equity-and-health-policy/fact-sheet/key-facts-on-health-coverage-of-immigrants"
+    )
 
     def formula(person, period, parameters):
         category = person("medicaid_category", period)
-        return category != category.possible_values.NONE
+        categorically_eligible = category != category.possible_values.NONE
+        istatus = person("immigration_status", period)
+        undocumented = istatus == istatus.possible_values.UNDOCUMENTED
```

Comment:
For posterity, we're merging this for now but https://github.com/PolicyEngine/policyengine-us/issues/3324